lib/node/node.js

```diff
@@ -337,15 +337,19 @@ module.exports.create = function createNode(type, expectedParams, options) {
             var validator = expectedParams[expectedParamName];
 
             if (!validator.nullable && !this.params.hasOwnProperty(expectedParamName)) {
-                throw new Error('Missing required param "' + expectedParamName + '"');
+                var requiredParamError = new Error('Missing required param "' + expectedParamName + '"');
+                requiredParamError.node_id = this.id();
+                throw requiredParamError;
             }
 
             if (!validator.isValid(param)) {
-                throw new Error(
+                var isNotValidError = new Error(
                     'Invalid type for param "' + expectedParamName + '",' +
                         ' expects "' + validator.expects + '" type,' +
                         ' got `' + JSON.stringify(param) + '`'
                 );
+                isNotValidError.node_id = this.id();
```

Comment:
This is the internal ID of the node; it's not the id the user provided. Exposing this id doesn't help the end user because the id the user provided is not this one.

We need to rethink the naming; probably we need to rename the internal node id to `node.ref` or something similar. Or the external to `node.name`.